src/github.com/matrix-org/dendrite/roomserver/types/types.go

```diff
@@ -13,13 +13,32 @@ type PartitionOffset struct {
 	Offset int64
 }
 
+// EventTypeNID is a numeric ID for an event type.
+type EventTypeNID int64
+
+// EventStateKeyNID is a numeric ID for an event state_key.
+type EventStateKeyNID int64
+
+// EventNID is a numeric ID for an event.
+type EventNID int64
+
+// RoomNID is a numeric ID for a room.
+type RoomNID int64
+
+// StateSnapshotNID is a numeric ID for the state at an event.
+type StateSnapshotNID int64
+
+// StateDataNID is a numeric ID for a block of state data.
+// These blocks of state data are combined to form the actual state.
+type StateDataNID int64
```

Comment:
Done 437f298

I think Entry is fairly common jargon for a key value pair in a map. So I'm inclined to keep that one.